### PR TITLE
[GNA] Fix import model with header version 2.1

### PR DESCRIPTION
**Pull request by admitriev-gna**

### CODE REVIEW

**Review by bsochack — APPROVED** [review]
Is there any possibility to add test?

**Review by aalbersk — APPROVED** [review]

**Comment by aalbersk** [comment]
> Is there any possibility to add test?

+1 on this one. If there'll be a possibility to have tests for all header versions, or at least the 1st one, it would be perfect

**Comment by aalbersk** [comment]
Did the 2.1 header work when 2.3 was the latest one? If no, then, the [constructor for 2.1 in 2.3](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/gna_plugin/serial/headers/2dot3/gna_model_header.hpp#L70) is redundant and incorrect and should be removed or fixed.

File: inference-engine/tests/functional/plugin/shared/include/behavior/test_model_repo.hpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    const char *getModelPathNonFatal() noexcept;
+
+    std::string get_data_path();
+
+    inline const char *getModelPathNonFatalDefault() noexcept {
+        if (const auto envVar = std::getenv("MODELS_PATH")) {
+            return envVar;
+        }
+
+#ifdef MODELS_PATH
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/shared/include/behavior/test_model_repo.hpp:21]: Why do you add `MODELS_PATH` from deprecated_tests? We refused to use these variables in new tests

**Review by iefode — CHANGES_REQUESTED** [review]

File: inference-engine/tests/functional/plugin/gna/backword_compatibility.cpp [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+std::string TestDataHelpers::get_data_path() {
+    return TestDataHelpers::get_data_path_default();
+}
+
+
+namespace LayerTestsDefinitions {
+
+class BackwardCompatibilityTest : public testing::WithParamInterface<exportImportNetworkParams>,
```
[the remark this answers is not in the document]
Comment (admitriev-gna) [reply inference-engine/tests/functional/plugin/gna/backword_compatibility.cpp:44]: Tests that are imported with versions 2.1 and 2.2 failing without changes.

File: inference-engine/tests/functional/plugin/gna/backword_compatibility.cpp [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        Infer();
+
+        const auto& actualOutputs = GetOutputs();
+        auto referenceOutputs = CalculateRefs();
+        Compare(referenceOutputs, actualOutputs);
+
+        for (auto const& configItem : importConfiguration) {
+            configuration[configItem.first] = configItem.second;
+        }
+        std::string inputTensorBinary = TestDataHelpers::get_data_path() + "/gna/" + nameExportModel;
```
[the remark this answers is not in the document]
Comment (admitriev-gna) [reply inference-engine/tests/functional/plugin/gna/backword_compatibility.cpp:80]: I created a pull request, where I added exported models with different versions: https://github.com/openvinotoolkit/testdata/pull/3

[the remark this answers is not in the document]
Comment (bsochack) [reply inference-engine/tests/functional/plugin/gna/backword_compatibility.cpp]: great. thanks!

**Review by bsochack — APPROVED** [review]

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/import_export_tests/backword_compatibility.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2020 Intel Corporation
```
Comment (dorloff) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/import_export_tests/backword_compatibility.cpp:1]: backwArd compatibility. Please correct.

**Review by iefode — APPROVED** [review]

**Pull request closed without merging**